bin/reth/src/test_eth_chain/runner.rs

```diff
@@ -1,13 +1,161 @@
+        return true
+    }
+    // txbyte is of type 02 and we dont parse tx bytes for this test to fail.
+    if path.file_name() == Some(OsStr::new("typeTwoBerlin.json")) {
+        return true
+    }
+    // Test checks if nonce overflows. We are handling this correctly but we are not parsing
+    // exception in testsuite There are more nonce overflow tests that are in internal
+    // call/create, and those tests are passing and are enabled.
+    if path.file_name() == Some(OsStr::new("CreateTransactionHighNonce.json")) {
+        return true
+    }
+
+    // Test check if gas price overflows, we handle this correctly but does not match tests specific
+    // exception.
+    if path.file_name() == Some(OsStr::new("HighGasPrice.json")) {
+        return true
+    }
+
+    // Skip test where basefee/accesslist/diffuculty is present but it shouldn't be supported in
+    // London/Berlin/TheMerge. https://github.com/ethereum/tests/blob/5b7e1ab3ffaf026d99d20b17bb30f533a2c80c8b/GeneralStateTests/stExample/eip1559.json#L130
+    // It is expected to not execute these tests.
+    if path.file_name() == Some(OsStr::new("accessListExample.json")) ||
+        path.file_name() == Some(OsStr::new("basefeeExample.json")) ||
+        path.file_name() == Some(OsStr::new("eip1559.json")) ||
+        path.file_name() == Some(OsStr::new("mergeTest.json"))
+    {
+        return true
+    }
+
+    // These tests are passing, but they take a lot of time to execute so we are going to skip them.
+    if path.file_name() == Some(OsStr::new("loopExp.json")) ||
+        path.file_name() == Some(OsStr::new("Call50000_sha256.json")) ||
+        path.file_name() == Some(OsStr::new("static_Call50000_sha256.json")) ||
+        path.file_name() == Some(OsStr::new("loopMul.json")) ||
+        path.file_name() == Some(OsStr::new("CALLBlake2f_MaxRounds.json"))
+    {
+        return true
+    }
+    false
+}
 
 /// Run one JSON-encoded Ethereum blockchain test at the specified path.
-pub async fn run_test(path: &Path) -> eyre::Result<()> {
+pub async fn run_test(path: PathBuf) -> eyre::Result<()> {
+    let path = path.as_path();
     let json_file = std::fs::read(path)?;
-    let suits: Test = serde_json::from_reader(&*json_file)?;
+    let suites: Test = serde_json::from_reader(&*json_file)?;
+
+    if should_skip(path) {
+        return Ok(())
+    }
+
+    for (name, suite) in suites.0 {
+        if matches!(
+            suite.network,
+            ForkSpec::ByzantiumToConstantinopleAt5 |
+                ForkSpec::Constantinople |
+                ForkSpec::MergeEOF |
+                ForkSpec::MergeMeterInitCode
+        ) {
+            continue
+        }
+
+        if matches!(suite.pre, State(RootOrState::Root(_))) {}
+
+        let pre_state = match suite.pre.0 {
+            RootOrState::State(state) => state,
+            RootOrState::Root(_) => {
+                info!("Skipping test {name}...");
+                continue
+            }
```

Comment:
Why do we skip these?

Reply:
Reusability of `RootOrState` (Used in post_state) , for pre it can be only State, will modify it properly.